### PR TITLE
Feature/graph chart

### PR DESCRIPTION
**Pull request by xyz8983**
add graph chart
update css

### CODE REVIEW

File: client/ts/graph_chart.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    let simulation = d3.forceSimulation()
+        .force('charge', d3.forceManyBody())
+        .force('center', d3.forceCenter(width/2, height/2))
+        // .force('collide', d3.forceCollide().radius(nodeRadius*1.2))
+        .force('collide', d3.forceCollide())
+        .force('link', d3.forceLink().distance(80)); // distance sets the length of each link
+
+    d3.json(jsonUrl.value, function(error, data){
+        console.log(data);
```
Comment (kanastasakis26) [inline client/ts/graph_chart.ts]: Is this needed?

File: client/ts/graph_chart.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        let links = svg.append("g")
+            .selectAll("line")
+            .data(data.links)
+            .enter()
+            .append("line")
+            .attr("stroke", "#FFFFFF")
```
Comment (kanastasakis26) [inline client/ts/graph_chart.ts]: Would be helpful to keep JS file with constants for the colors?

**Review by kanastasakis26 — APPROVED** [review]

**Review by boyoonc — APPROVED** [review]

File: client/ts/graph_chart.ts (hunk 1 above)
Comment (xyz8983) [reply client/ts/graph_chart.ts]: nope...

File: client/ts/graph_chart.ts (hunk 2 above)
Comment (xyz8983) [reply client/ts/graph_chart.ts]: good suggestions! I will create a color theme file

File: client/ts/graph_chart.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+export function generateGraghChart(jsonUrl){
+
+    // let svg = d3.select('svg'),
+    //     width = document.getElementById("drawing-section").clientWidth,
+    //     height = document.getElementById("drawing-section").clientHeight,
+    //     nodeRadius = 10;
+    // svg.attr("width", width).attr("height", height);
+    let svg = d3.select('svg'),
+        height = +svg.attr("height"),
```
Comment (besseym) [inline client/ts/graph_chart.ts:12]: `= +`?

**Review by besseym — APPROVED** [review]
👍

File: client/ts/graph_chart.ts (hunk 3 above)
Comment (xyz8983) [reply client/ts/graph_chart.ts:12]: to make sure the height is positive value

Comment (todddeluca) [reply client/ts/graph_chart.ts:12]: Neat!

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_plus_2

**Pull request merged**